ansible_base/lib/redis/client.py

```diff
@@ -113,29 +134,30 @@ def _get_hosts(self) -> None:
         self.connection_settings.pop('port', None)
         self.connection_settings['startup_nodes'] = []
 
-        translated_generic_exception = ImproperlyConfigured(_('Unable to parse redis_hosts, see logs for more details'))
-
         # Make sure we have a string for redis_hosts
-        if not isinstance(self.redis_hosts, str):
-            logger.error(f"Specified redis_hosts is not a string, got: {self.redis_hosts}")
-            raise translated_generic_exception
-
-        host_ports = self.redis_hosts.split(',')
-        for host_port in host_ports:
-            try:
-                node, port_string = host_port.split(':')
-            except ValueError:
-                logger.error(f"Specified cluster_host {host_port} is not valid; it needs to be in the format <host>:<port>")
-                raise translated_generic_exception
-
-            # Make sure we have an int for the port
-            try:
-                port = int(port_string)
-            except ValueError:
-                logger.error(f'Specified port on {host_port} is not an int')
-                raise translated_generic_exception
-
-            self.connection_settings['startup_nodes'].append(ClusterNode(node, port))
+        try:
+            if not isinstance(self.redis_hosts, str):
+                logger.error(f"Specified redis_hosts is not a string, got: {self.redis_hosts}")
+                raise ValueError()
```

Comment:
Took me a little bit to find the value error that the new except is trying to catch.

Consider using try/except/else and move the except that was added below to up here.

https://www.geeksforgeeks.org/try-except-else-and-finally-in-python/

Reply:
Talked with @john-westcott-iv offline. I was wrong with this suggestion.